include/crab/domains/symbolic_variable_eq_domain.hpp

```diff
@@ -0,0 +1,1031 @@
+      bool keep_rep = false;
+      for (auto &e : s) {
+        if (e == rep) {
+          keep_rep = true;
+          break;
+        }
+      }
+      element_t new_rep = keep_rep ? rep : s[0];
+      for (auto &e : s) {
+        res.m_parents.insert({e, new_rep});
+      }
+      equivalence_class_t ec = m_classes.at(rep);
+      res.m_classes.insert({new_rep, ec});
+    }
+    std::swap(res, *this);
+    normalize();
+    CRAB_LOG("symb-var-eq", crab::outs() << "After projection "; dump(););
+  }
+
+  void rename(const std::vector<element_t> &old_elements,
+              const std::vector<element_t> &new_elements) {
+    SVEQ_DOMAIN_SCOPED_STATS(".rename");
+
+    if (is_top() || is_bottom()) {
+      return;
+    }
+    if (old_elements.size() != new_elements.size()) {
+      CRAB_ERROR(domain_name(),
+                 "::rename with input vectors of different sizes");
+    }
+    for (unsigned i = 0, size = old_elements.size(); i < size; ++i) {
+      const element_t &old_v = old_elements[i];
+      const element_t &new_v = new_elements[i];
+      if (!contains(old_v)) {
+        CRAB_ERROR(domain_name(), "::rename assumes that ", old_v, " exists");
+      }
+      if (contains(new_v)) {
+        CRAB_ERROR(domain_name(), "::rename assumes that ", new_v,
+                   " does not exist");
+      }
+      boost::optional<element_t> parent_old_v;
+      for (auto it = m_parents.begin(), et = m_parents.end(); it != et;) {
+        if ((*it).first == old_v) {
+          // find it as key, remove it
+          parent_old_v = (*it).second;
+          it = m_parents.erase(it);
+        } else if ((*it).second == old_v) {
+          // find it as value
+          (*it).second = new_v;
+          ++it;
+        } else {
+          ++it;
+        }
+        if (parent_old_v) {
+          // insert a key-value pair for new_v
+          m_parents.insert(
+              {new_v, (*parent_old_v == old_v ? new_v : *parent_old_v)});
+          auto it = m_classes.find(old_v);
+          if (it != m_classes.end()) {
+            // update m_classes
+            equivalence_class_t ec = it->second;
+            m_classes.erase(it);
+            m_classes.insert({new_v, std::move(ec)});
+          }
+        }
+      }
+    }
+  }
+
+  /// @brief A normalization function used for other domain operations
+  /// @note  perform normalization to eliminate any class such as:
+  ///          {v3}=>#var5
+  void normalize() {
+    SVEQ_DOMAIN_SCOPED_STATS(".normalize");
+    // FIXME: change this
+    return;
+
+    CRAB_LOG("symb-var-eq", crab::outs()
+                                << "Before normalization " << *this << "\n";);
+    // TODO: could implement to check each size of class instead of creating
+    // equivalence_class_elems_t map
+    equivalence_class_elems_t equiv_classes = equiv_classes_elems();
+    for (auto &kv : equiv_classes) {
+      const element_t &rep = kv.first;
+      const element_set_t &s = kv.second;
+      if (s.size() == 1) {
```

Comment:
This is not what we talked about. If our equality domains are spread out, the value of the class matters even if the class is of size 1